LogicalTypes.md

```diff
@@ -366,3 +366,45 @@ optional group my_map (MAP_KEY_VALUE) {
 }
 ```
 
+### Unions
+
+A Union type annotates data stored as a Group.
+It describes the different possible types under the same field name.
+The group contains one optional field for each possible type in the Union.
+The names of the fields in the annotated Group are not important, but as a convention the type names are used.
+
+#### Nullability
+ - If the union is not nullable then exactly one field is non-null and the field containing the union is required.
+```
+// Union<String, Integer, Boolean>
+// (exactly one of either String, Integer or Boolean is non-null)
+required group my_union (UNION) {
+  optional binary string (UTF8);
+  optional int32 integer;
+  optional boolean bool;
+}
+```
+A projection might return an empty union if the non-null field is projected out. However we know that the Union is non-null,
+it just contains a value that was not read from disk.
+
+ - If the union is nullable then at most one field is non-null and the field containing the union is optional
+```
+// Optional<Union<String, Integer, Boolean>>
```

Comment:
I think it would be more clear if this was `Union<String, Integer, Boolean>` and noted "the value of the union may be null" rather than adding the `Optional` level. That makes it clear that it isn't actually wrapped, it is just that the value can be null in this case and that is encoded by the definition level.